frontend/src/components/data-table/column-header.tsx

```diff
@@ -54,37 +80,47 @@ export const DataTableColumnHeader = <TData, TValue>({
   }
 
   return (
-    <DropdownMenu modal={false}>
-      <DropdownMenuTrigger asChild={true}>
-        <div
-          className={cn(
-            "group flex items-center my-1 space-between w-full select-none gap-2 border hover:border-border border-transparent hover:bg-[var(--slate-3)] data-[state=open]:bg-[var(--slate-3)] data-[state=open]:border-border rounded px-1 -mx-1",
-            className,
-          )}
-          data-testid="data-table-sort-button"
-        >
-          <span className="flex-1">{header}</span>
-          <span
+    <>
+      <DropdownMenu modal={false}>
+        <DropdownMenuTrigger asChild={true}>
+          <div
             className={cn(
-              "h-5 py-1 px-1",
-              !column.getIsSorted() &&
-                "invisible group-hover:visible data-[state=open]:visible",
+              "group flex items-center my-1 space-between w-full select-none gap-2 border hover:border-border border-transparent hover:bg-[var(--slate-3)] data-[state=open]:bg-[var(--slate-3)] data-[state=open]:border-border rounded px-1 -mx-1",
+              className,
             )}
+            data-testid="data-table-sort-button"
           >
-            {renderSortIcon(column)}
-          </span>
-        </div>
-      </DropdownMenuTrigger>
-      <DropdownMenuContent align="start">
-        {renderDataType(column)}
-        {renderSorts(column)}
-        {renderCopyColumn(column)}
-        {renderColumnPinning(column)}
-        {renderColumnWrapping(column)}
-        {renderFormatOptions(column)}
-        <DropdownMenuItemFilter column={column} />
-      </DropdownMenuContent>
-    </DropdownMenu>
+            <span className="flex-1">{header}</span>
+            <span
+              className={cn(
+                "h-5 py-1 px-1",
+                !column.getIsSorted() &&
+                  "invisible group-hover:visible data-[state=open]:visible",
+              )}
+            >
+              {renderSortIcon(column)}
+            </span>
+          </div>
+        </DropdownMenuTrigger>
+        <DropdownMenuContent align="start">
+          {renderDataType(column)}
+          {renderSorts(column)}
+          {renderCopyColumn(column)}
+          {renderColumnPinning(column)}
+          {renderColumnWrapping(column)}
+          {renderFormatOptions(column)}
+          <DropdownMenuItemFilter column={column} />
+          <RenderSetFilter onClick={() => setIsSetFilterOpen(true)} />
```

Comment:
instead of `onClick`, what about as a `<DropdownMenuSub` like we do with the other filter